llama-index-core/llama_index/core/instrumentation/__init__.py

```diff
@@ -31,3 +39,45 @@ def get_dispatcher(name: str = "root") -> Dispatcher:
     )
     root_manager.add_dispatcher(new_dispatcher)
     return new_dispatcher
+
+
+class DispatcherSpanMixin(ABC):
+    """
+    Apply the `dispatcher.span` decorator to implementations of abstract methods, as well
+    as any methods previously decorated (in any base class) that are being overridden by
+    a subclass. For example, if class `A` has abstract method `f`, and class `B` inherits
+    from `A` and provides an implementation of `f`, then `B.f` will be decorated by the mixin.
+    Furthermore, if `B` has a non-abstract method `g` that is decorated by `dispatcher.span`
+    and new class `C` inherits from `B` and overrides `g`, then `C.g` will also be decorated
+    by the mixin. Note that users can still manually apply `dispatcher.span` to the methods
+    in their custom subclasses without creating duplicate spans because the `dispatcher.span`
+    decorator should be idempotent.
```

Comment:
I fully follow now. Wondering if we can maybe add a note for future devs that we achieve idempotency thru `DISPATCHER_SPAN_DECORATED_ATTR` and the check within `dispatcher.span` that utilizes it?

Without knowing this then the statement on L82 `if attr in abstract_methods or attr in decorated_methods` would make it seem like we're dispatching a new span multiple times (i,.e `attr in decorated_methods`).

Reply:
Please ignore this comment. I see you've left a comment in the check within the logic for `dispatcher.span()`.